.github/workflows/build.yml

```
@@ -74,31 +92,47 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-ccache-${{ github.head_ref }}
           ${{ runner.os }}-ccache
```

```suggestion
          ${{ matrix.os }}-ccache
```
Both Ubuntu jobs are sharing the same cache currently.

---

I suppose we need to use the more specific matrix.os instead of runner.os for all cache keys. Done.